datafusion/execution/src/memory_pool/pool.rs

```diff
@@ -145,21 +145,21 @@ impl MemoryPool for FairSpillPool {
 
     fn unregister(&self, consumer: &MemoryConsumer) {
         if consumer.can_spill {
-            self.state.lock().num_spill -= 1;
+            self.state.lock().num_spill.checked_sub(1).unwrap();
```

Comment:
Drive by sanity check (as first version of `MemoryReservation::split` would unregister the same consumer multiple times :sweat_smile:

Reply:
Maybe it would be worth adding some unit tests to the `MemoryReservation` now given it is growing in sophistication